src/Symfony/Component/HttpFoundation/Tests/RequestTest.php

```diff
@@ -78,14 +81,27 @@ public function testIsNoCache()
         $this->assertFalse($isNoCache);
     }
 
+    /**
+     * @group legacy
+     */
     public function testGetContentType()
     {
+        $this->expectDeprecation('Since symfony/http-foundation 6.1: The method "Symfony\Component\HttpFoundation\Request::getContentType" is deprecated, use "getContentTypeFormat" instead.');
         $request = new Request();
+
         $contentType = $request->getContentType();
 
         $this->assertNull($contentType);
     }
 
+    public function testGetContentTypeFormat()
+    {
+        $request = new Request();
+        $contentType = $request->getContentTypeFormat();
+
+        $this->assertNull($contentType);
```

Comment:
Would be great to add more tests covering cases where that method returns a non-null value. Otherwise, we don't really have tests for the feature.

Reply:
Thank you for the suggestion. I added tests to cover the mime types `application/json` and `text/html`